src/physics/em/KleinNishinaModel.hh

```diff
@@ -9,7 +9,7 @@
 
 #include "physics/base/Model.hh"
 #include "physics/base/ParticleParams.hh"
-#include "detail/KleinNishina.hh"
+#include "detail/KleinNishinaInteractor.hh"
```

Comment:
I apologize in advance for my faulty memory, but was I the one who suggested rolling all the "interactor" data classes into the `Interactor.hh` header itself? It reduces the number of files (which is admittedly quite large) but exposes all the interactor implementation detail to downstream components (viz., the ones that create the models). 

All this to say: as an *ideal* we should separate the "data" definitions from the "operations" by only including the data types in this header, but I'm willing to accept lumping the two together for the sake of expediency.

Reply:
@sethrj No, I don't think that was discussed. By interactor data classes, I assume you're talking about classes like KleinNishinaPointers? I ended up moving those to the *Interactor.hh files because of circular dependencies in headers. Happy to move them somewhere else / put them in a new file if you'd like though.

Reply:
@paulromano I think we should be able to do something like this without any circular dependencies. This way:
- CPU code (Model bas class) stays out of GPU code
- Interactor can be tested independently of physics implementation, RNG choice, etc.
- Code instantiating the models only gets propagated the data components of the interactor, rather than all the low-level implementation components (stack allocator, particle track view)

The downside of course is that it means we have *five* user-created files and *three* computer-generated files for every physics model, which can make the code harder to organize. Do you think the trade-off is worth it? Do you think adding another level of subdirectories would help or hurt?

![em-kernel](https://user-images.githubusercontent.com/741229/135930598-e9a5ce96-51b6-4363-8156-678ce5cce2fb.png)

Reply:
Just to reiterate our offline discussion, I do think this organization is an improvement over what I had currently. I've gone ahead and implemented this layout.